compiler/rustc_ast_lowering/src/format.rs

```diff
@@ -223,6 +223,9 @@ fn expand_format_args<'hir>(
     // in order, we can use a simple array instead of a `match` construction.
     // However, if there's a yield point in any argument except the first one,
     // we don't do this, because an ArgumentV1 cannot be kept across yield points.
+    //
+    // This is an optimization, speeding up compilation about 1-2% in some cases.
+    // See https://perf.rust-lang.org/compare.html?start=5dbee4d3a6728eb4530fb66c9775834438ecec74&end=e36affffe97378a0027b4bcfbb18d27356164ed0&stat=instructions:u
```

Comment:
I'm not sure how much of a permalink that is. Older perf results might get inaccessible in the future. I think it would be better to link to the github comment with the perf results instead, that contains a summary that is preserved, plus the discussion which lets you derive key facts from the context even if the perf run link itself is gone.

Reply:
The entire summary is basically "speeding up compilation about 1-2% in some cases", which is included here.

Even if the link dies, the commit hashes will still be available, so it'd be easy to run the benchmarks again.

Reply:
I've checked perf runs of older PRs, many older ones are gone.

Gone (2017): https://github.com/rust-lang/rust/pull/45319#issuecomment-336907788

Gone (2019): https://github.com/rust-lang/rust/pull/60630#issuecomment-490561552

Still there (2020): https://github.com/rust-lang/rust/pull/78323#issuecomment-716140911

I'd prefer a link like: https://github.com/rust-lang/rust/pull/106745#issuecomment-1380268745

As for re-running benchmarks, I'm not that sure that it's this easy. In the time windows we are talking about (3-4 years), MSRV support of the benchmark suite, docker CI image and glibc version requirements start to play a role. It's probably possible but I doubt it is possible with the press of a button.

Reply:
Updated the link to point to a github comment from the timer bot.